runners/pvc_autoresizer_test.go

```diff
@@ -81,4 +89,143 @@ var _ = Describe("test resizer", func() {
 			}
 		})
 	})
+
+	Context("resize", func() {
+		Context("parameter tests", func() {
+			ctx := context.Background()
+			pvcNS := "default"
+			increase := "1Gi"
+			limit := int64(100 << 30)
+			volumeMode := corev1.PersistentVolumeFilesystem
+
+			testCases := []struct {
+				description   string
+				pvcSizeGi     int64
+				expectSizeGi  int64
+				threshold     string
+				availableByte int64
+			}{
+				{
+					description:   "Should resize(absolute value)",
+					pvcSizeGi:     10,
+					expectSizeGi:  11,
+					threshold:     "5Gi",
+					availableByte: 5<<30 - 1,
+				},
+				{
+					description:   "Should not resize(absolute value)",
+					pvcSizeGi:     10,
+					expectSizeGi:  10,
+					threshold:     "5Gi",
+					availableByte: 5 << 30,
+				},
+				{
+					description:   "Should resize(%)",
+					pvcSizeGi:     10,
+					expectSizeGi:  11,
+					threshold:     "50%",
+					availableByte: 5<<30 - 1,
+				},
+				{
+					description:   "Should not resize(%)",
+					pvcSizeGi:     10,
+					expectSizeGi:  10,
+					threshold:     "50%",
+					availableByte: 5 << 30,
+				},
+			}
+
+			counter := 0
+			for _, tc := range testCases {
+				counter++
+				pvcName := fmt.Sprintf("test-pvc-%d", counter)
+				pvcSizeGi := tc.pvcSizeGi
+				expectSizeGi := tc.expectSizeGi
+				threshold := tc.threshold
+				availableByte := tc.availableByte
+
+				description := fmt.Sprintf(
+					"%s: pvcSizeGi=%d expectSizeGi=%d threshold=%s availableByte=%d",
+					tc.description,
+					tc.pvcSizeGi,
+					tc.expectSizeGi,
+					tc.threshold,
+					tc.availableByte)
+
+				It(description, func() {
+					createPVC(ctx, pvcNS, pvcName, scName, threshold, increase, pvcSizeGi<<30, limit, volumeMode)
+					setMetrics(pvcNS, pvcName, availableByte, pvcSizeGi<<30)
+					testFunc := func() error {
+						var pvc corev1.PersistentVolumeClaim
+						err := k8sClient.Get(ctx, types.NamespacedName{Namespace: pvcNS, Name: pvcName}, &pvc)
+						if err != nil {
+							return err
+						}
+						req := pvc.Spec.Resources.Requests.Storage().Value()
+						if req>>30 != expectSizeGi {
```

Comment:
```suggestion
						ALLOWANCE := 1 << 10
						if !(expectSizeGi << 30 - ALLOWANCE < req && req <= expectSizeGi << 30 + ALLOWANCE)  {
```

Reply:
Thank you. I improved it.
https://github.com/topolvm/pvc-autoresizer/pull/48/commits/0d7c4c2b5648fa364809c11ec781308eea0d94e8